src/goto-programs/character_refine_preprocess.h

```diff
@@ -0,0 +1,153 @@
+  static exprt expr_of_is_surrogate(const exprt &chr, const typet &type);
+  static void convert_is_surrogate(conversion_input &target);
+  static void convert_is_surrogate_pair(conversion_input &target);
+  static exprt expr_of_is_title_case(const exprt &chr, const typet &type);
+  static void convert_is_title_case_char(conversion_input &target);
+  static void convert_is_title_case_int(conversion_input &target);
+  static exprt expr_of_is_letter_number(const exprt &chr, const typet &type);
+  static exprt expr_of_is_unicode_identifier_part(
+    const exprt &chr, const typet &type);
+  static void convert_is_unicode_identifier_part_char(conversion_input &target);
+  static void convert_is_unicode_identifier_part_int(conversion_input &target);
+  static exprt expr_of_is_unicode_identifier_start(
+    const exprt &chr, const typet &type);
+  static void convert_is_unicode_identifier_start_char(
+    conversion_input &target);
+  static void convert_is_unicode_identifier_start_int(conversion_input &target);
+  static exprt expr_of_is_ascii_upper_case(const exprt &chr, const typet &type);
+  static void convert_is_upper_case_char(conversion_input &target);
+  static void convert_is_upper_case_int(conversion_input &target);
+  static exprt expr_of_is_valid_code_point(const exprt &chr, const typet &type);
+  static void convert_is_valid_code_point(conversion_input &target);
+  static exprt expr_of_is_whitespace(const exprt &chr, const typet &type);
+  static void convert_is_whitespace_char(conversion_input &target);
+  static void convert_is_whitespace_int(conversion_input &target);
+  static exprt expr_of_low_surrogate(const exprt &chr, const typet &type);
+  static void convert_low_surrogate(conversion_input &target);
+  static exprt expr_of_reverse_bytes(const exprt &chr, const typet &type);
+  static void convert_reverse_bytes(conversion_input &target);
+  static exprt expr_of_to_chars(const exprt &chr, const typet &type);
+  static void convert_to_chars(conversion_input &target);
+  static void convert_to_code_point(conversion_input &target);
+  static exprt expr_of_to_lower_case(const exprt &chr, const typet &type);
+  static void convert_to_lower_case_char(conversion_input &target);
+  static void convert_to_lower_case_int(conversion_input &target);
+  static exprt expr_of_to_title_case(const exprt &chr, const typet &type);
+  static void convert_to_title_case_char(conversion_input &target);
+  static void convert_to_title_case_int(conversion_input &target);
+  static exprt expr_of_to_upper_case(const exprt &chr, const typet &type);
+  static void convert_to_upper_case_char(conversion_input &target);
+  static void convert_to_upper_case_int(conversion_input &target);
+
+  // Helper functions
+  static void convert_char_function(
+      exprt (*expr_function)(const exprt &chr, const typet &type),
+      conversion_input &target);
+  static exprt in_interval_expr(
+    const exprt &chr, mp_integer lower_bound, mp_integer upper_bound);
```

Comment:
There is some mixed indent of parameters - either seems ok, but consistency would be great.